src/robot_state_publisher.cpp

```diff
@@ -175,14 +175,14 @@ void RobotStatePublisher::setupURDF(const std::string & urdf_xml)
 
   // Initialize the mimic map
   mimic_.clear();
-  for (const std::pair<std::string, urdf::JointSharedPtr> & i : model_->joints_) {
```

Comment:
IIUC another way to fix would be to make the type match what the compiler expects. It looks like it expects the first element in the pair to be `const std::string`.

```
/home/jenkins-agent/workspace/nightly_linux_clang_libcxx/ws/src/ros/robot_state_publisher/src/robot_state_publisher.cpp:178:61: warning: loop variable 'i' has type 'const std::pair<std::string, urdf::JointSharedPtr> &' (aka 'const pair<basic_string<char, char_traits<char>, allocator<char> >, shared_ptr<urdf::Joint> > &') but is initialized with type 'std::__1::__map_iterator<std::__1::__tree_iterator<std::__1::__value_type<std::__1::basic_string<char>, std::__1::shared_ptr<urdf::Joint> >, std::__1::__tree_node<std::__1::__value_type<std::__1::basic_string<char>, std::__1::shared_ptr<urdf::Joint> >, void *> *, long> >::value_type' (aka 'pair<const std::__1::basic_string<char>, std::__1::shared_ptr<urdf::Joint> >') resulting in a copy [-Wrange-loop-construct]
```
If I'm reading the difference between the two types correctly

```c++
// Loop variable type
const pair<basic_string<char, char_traits<char>, allocator<char> >, shared_ptr<urdf::Joint> > &
// Actual type returned by the map iterator
pair<const std::__1::basic_string<char>, std::__1::shared_ptr<urdf::Joint> >
```

Reply:
I don't mind either way. 94b6f42a10094edb7827502230df314b98ae34c4 maintains the explicit type without using `auto`: [![Build Status](https://ci.ros2.org/buildStatus/icon?job=ci_linux&build=11469)](https://ci.ros2.org/job/ci_linux/11469/) (only unrelated warnings remaining).

Reply:
Yeah, I like 94b6f42 better.  Looks good to me, thanks for the update.